utils/sampler/README.md

```diff
@@ -0,0 +1,95 @@
+editing the files you want to edit.
+
+```
+Hint: Before you select a file in the app, if you want to change the template you
+are inserting into, change it in the original dartdoc comments before you
+extract.
+```
+
+When the app starts, type part of the name of the framework file in which the
+sample would like to edit resides, and it will autocomplete. Select the file,
+and it will show you a list of the available elements to edit the sample on, or
+to add a new sample to. The app monitors the filesystem, so if the selected
+framework file changes, it will re-parse the file.
+
+You can filter which elements are shown by unchecking the type of elements you
+wish to filter out, and you can sort alphabetically, or by line number.
+
+If you want to add a new sample, click the "ADD SAMPLE" button next to the
+element you want to add the sample to. The sample will be added at the end of
+the documentation comment, just before the "See also:" section, if any. The
+inserted sample will be immediately added to the actual framework file, and
+`sampler` will select the new sample for you to edit.
+
+Once you add a new sample, or select a sample with the `SELECT` button, it will
+move to a sample detail page where you can extract that sample into a temporary
+project, and open the project in an IDE or editor.
+
+Once on the sample page, you can extract the sample with the `EXTRACT SAMPLE`
+button. It will create a temporary project and prepare it for editing.
+
+Once you have extracted the sample, you can copy the path to its main and open
+that in your editor, or use the `OPEN IN VS CODE` or `OPEN IN INTELLIJ` buttons
+to open then project in an IDE.
+
+Now edit the selected sample.  Be careful to only edit the sections inside the
+section markers: any editing occurring outside the section markers will be lost
+when the code is reinserted. If you mangle the section markers themselves, it
+probably will fail to re-insert. If you need to make a change to code that is
+not part of a section, you will have to edit the template, or create a new
+template instead.
+
+Here's what the section markers look like:
+
+```
+//********************************************************************
+//* ▼▼▼▼▼▼▼▼ code ▼▼▼▼▼▼▼▼ (do not modify or remove section marker)
+
+//[ code goes here ...]
+
+//* ▲▲▲▲▲▲▲▲ code ▲▲▲▲▲▲▲▲ (do not modify or remove section marker)
+//********************************************************************
+```
+
+After you're done with your modifications to the sample, save the code to disk,
+and then go back to the `sampler` app.  You can then click the `REINSERT` button
+to place edited sample back into the documentation comments in the original
+framework file. You can open that file from the `sampler` app with the
+appropriate buttons, or copy/paste the path into your editor.
+
+## Missing features
+
+This is still a work in progress, so if you find something missing, either [send
```

Comment:
Might want to mention something about code signing issues on macOS?

Reply:
Did you have code signing issues? I turned off the sandboxing, so it should be running without needing code signing if you run it with the script (`flutter run`).

Reply:
I didn't try it with this PR. This was just my experience from when you first pointed us at the tool. Is turning off the sandboxing something the app can do, or does every user need to turn it off manually?

Reply:
When you `flutter run` it you're building it from scratch like any Apple developer would, so the app can turn off the sandboxing in the build settings.